### ISSUE TITLE
[🐛 bug report] hdrfilm.cpp: duplicate channel detection

### ISSUE THREAD

**Issue by Andy3531** [mitsuba-renderer/mitsuba2#629]
- [🐛 bug report]

## Description
Here is the original code.
```
    void prepare(const std::vector<std::string> &channels) override {
        std::vector<std::string> channels_sorted = channels;
        channels_sorted.push_back("R");
        channels_sorted.push_back("G");
        channels_sorted.push_back("B");
        std::sort(channels_sorted.begin(), channels_sorted.end());
        for (size_t i = 1; i < channels.size(); ++i) {
            if (channels[i] == channels[i - 1])
                Throw("Film::prepare(): duplicate channel name \"%s\"", channels[i]);
        }

        m_storage = new ImageBlock(m_crop_size, channels.size());
        m_storage->set_offset(m_crop_offset);
        m_storage->clear();
        m_channels = channels;
    }
```
And I think that it should be
```
        for (size_t i = 1; i < channels_sorted.size(); ++i) {
            if (channels_sorted[i] == channels_sorted[i - 1])
                Throw("Film::prepare(): duplicate channel name \"%s\"", channels_sorted[i]);
        }
```
in the function definition.

**Comment by njroussel** [comment]
Indeed, thank you for pointing this out :+1: .
I've fixed it in ec0200a681caa3c13d009caddf643de2777ea9cd

**Comment by njroussel** [comment]
:sweat_smile: thanks ...

FYI, we've released [Mitsuba 3](https://github.com/mitsuba-renderer/mitsuba3) and will be deprecating Mitsuba 2. I'd highly recommend you migrate your project(s) to the newer version if possible, as we'll actively be fixing issues/bugs and adding new features to it.
